internal/controller/connector_controller.go

```diff
@@ -0,0 +1,328 @@
+	if err := r.Get(ctx, templateConfigMapNamespacedName, &templateConfigMap); err != nil {
+		logger.Error(err, "fail to fetch ConfigMap which stores ConnectorTemplate config")
+
+		return nil, err
+	}
+
+	var templateConfig map[string]interface{}
+	err := json.Unmarshal([]byte(templateConfigMap.Data["config"]), &templateConfig)
+	if err != nil {
+		logger.Error(err, "fail to unmarshal ConnectorTemplate config")
+
+		return nil, err
+	}
+
+	var configs []map[string]interface{}
+
+	for _, stream := range connector.Spec.Streams {
+		config := make(map[string]interface{})
+		for k, v := range templateConfig {
+			config[k] = v
+		}
+
+		config["stream"] = stream
+
+		var patches map[string]map[string]interface{}
+		if connector.Spec.Patches != nil {
+			err = json.Unmarshal(connector.Spec.Patches, &patches)
+			if err != nil {
+				logger.Error(err, "fail to unmarshal Connector patches")
+
+				return nil, err
+			}
+		}
+
+		if connector.Spec.Patches != nil {
+			if val, ok := patches[stream]; ok {
+				for k, v := range val {
+					config[k] = v
+				}
+			}
+		}
+
+		configs = append(configs, config)
+	}
+
+	return configs, nil
+}
+
+func (r *ConnectorReconciler) createConnectorDeployment(ctx context.Context, connector v1beta1.Connector, stream, configMapName string) error {
+	name := v1beta1.GenConnectorDeploymentName(connector.Name, stream)
+	containerPorts := append(
+		[]corev1.ContainerPort{
+			{
+				ContainerPort: v1beta1.ConnectorContainerPortMap[connector.Spec.Type],
+			},
+		},
+		connector.Spec.ContainerPorts...,
+	)
+	deployment := appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: connector.Namespace,
+			Name:      name,
+			Labels: map[string]string{
+				hapi.ComponentKey: v1beta1.ComponentTypeConnector,
+				hapi.InstanceKey:  connector.Name,
+			},
+		},
+		Spec: appsv1.DeploymentSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					hapi.ComponentKey: v1beta1.ComponentTypeConnector,
+					hapi.InstanceKey:  connector.Name,
+					"stream":          stream,
+				},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						hapi.ComponentKey: v1beta1.ComponentTypeConnector,
+						hapi.InstanceKey:  connector.Name,
+						"stream":          stream,
+					},
+					Annotations: r.getPromAnnotations(connector),
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:  connector.Name,
+							Image: v1beta1.ConnectorImageMap[connector.Spec.Type],
+							Args: []string{
+								"run",
+								"--config /data/config/config.json",
+							},
+							Ports: containerPorts,
+							VolumeMounts: []corev1.VolumeMount{
+								{
+									Name:      configMapName,
+									MountPath: "/data/config",
+								},
+								{
+									Name:      "data",
+									MountPath: "/data",
+								},
+							},
+						},
+						{
+							Name:  "log",
+							Image: "busybox:1.36",
```

Comment:
I think we need to consider the situation where users cannot access Docker Hub, but this is not our current top priority and can be considered as a future to-do item.